### PR TITLE
feat: migrating login activity to mvvm

### PR DESCRIPTION
**Pull request by PratyushSingh07**
Jira ticket : https://mifosforge.jira.com/browse/MOBILE-67

Please Add Screenshots If there are any UI changes.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ ] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [ ] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [ ] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: app/src/main/java/org/mifos/mobile/viewModels/LoginViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+import org.mifos.mobile.models.Page
+import org.mifos.mobile.models.User
+import org.mifos.mobile.models.client.Client
+import org.mifos.mobile.models.payload.LoginPayload
+import org.mifos.mobile.utils.Constants
+import retrofit2.HttpException
+import java.net.UnknownHostException
+import javax.inject.Inject
+
+class LoginViewModel @Inject constructor(private val dataManager: DataManager?) : ViewModel() {
```
Comment (jawidMuhammadi) [inline app/src/main/java/org/mifos/mobile/viewModels/LoginViewModel.kt:22]: In MVVM architecture, don't we have a Repository as well??

File: app/src/main/java/org/mifos/mobile/viewModels/LoginViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    var success = MutableLiveData<String>()
+    val navigateToPassCodeActivity = MutableLiveData<String>()
+    val userName = MutableLiveData<String>()
+    val Password = MutableLiveData<String>()
+
+
+    fun login(loginPayload: LoginPayload?) {
+        if (isCredentialsValid(loginPayload)) {
+            compositeDisposable.add(
+                dataManager?.login(loginPayload)
```
Comment (jawidMuhammadi) [inline app/src/main/java/org/mifos/mobile/viewModels/LoginViewModel.kt:35]: We were supposed to use Coroutines, right?

File: app/src/main/java/org/mifos/mobile/viewModels/LoginViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+import org.mifos.mobile.utils.Constants
+import retrofit2.HttpException
+import java.net.UnknownHostException
+import javax.inject.Inject
+
+class LoginViewModel @Inject constructor(private val dataManager: DataManager?) : ViewModel() {
+    private val preferencesHelper: PreferencesHelper? = dataManager?.preferencesHelper
+    private val compositeDisposable: CompositeDisposable = CompositeDisposable()
+    var error = MutableLiveData<String>()
+    var success = MutableLiveData<String>()
```
Comment (jawidMuhammadi) [inline app/src/main/java/org/mifos/mobile/viewModels/LoginViewModel.kt:26]: Use a sealed class named as `LoginUiState`

**Review by jawidMuhammadi — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/mifos/mobile/viewModels/LoginViewModel.kt (hunk 2 above)
Comment (PratyushSingh07) [reply app/src/main/java/org/mifos/mobile/viewModels/LoginViewModel.kt:35]: yes jawid , but during the review of sprint 1 , rajan asked to continue with RxJava for the time being

File: app/src/main/java/org/mifos/mobile/viewModels/LoginViewModel.kt (hunk 1 above)
Comment (PratyushSingh07) [reply app/src/main/java/org/mifos/mobile/viewModels/LoginViewModel.kt:22]: yes we do and I had a doubt as to how we are fetching the data from the source incase of login activity

Comment (jawidMuhammadi) [reply app/src/main/java/org/mifos/mobile/viewModels/LoginViewModel.kt:22]: I'm unsure I understand what you mean by "source". However, I recommend creating a single repository called `UserAuthRepository`, including the logic for login and registration, and then using it in both the Register and Login view models.

**Comment by jawidMuhammadi** [comment]
Closing this as @gururani-abhishek has 3 related PRs. @gururani-abhishek  Let's create a UserAuthRepository for all the authentication-related data layers. Make only one PR for registration, login, and update password, and close your other PRs.

**Pull request closed without merging**